File: dev/20181026/FizzBuzz.rb
# frozen_string_literal: true

# FizzBuzzClass
class FizzBuzz
  def self.generate(number)
    value = 'Fizz' if number % 3 == 0
    value = value.to_s + 'Buzz' if number % 5 == 0
    value = number if value.nil?

    value
  end

  def self.iterate(count)
    array = []
    (1..count).each do |num|
      array.push(FizzBuzz.generate(num).to_s)
    end

    array
  end
end

File: dev/20181228/k2works/src/fizzbuzz.js
function iterate(count) {
  let array = [];
  [...Array(count + 1).keys()].forEach(function (n) {
    array.push(fizzBuzz(n));
  });
  return array;
}

function fizzBuzz(number) {
  let value = number;

  if ((number % 3 === 0) && (number % 5 === 0)) {
    value = "FizzBuzz";
  } else if (number % 3 === 0) {
    value = "Fizz";
  } else if (number % 5 === 0) {
    value = "Buzz";
  }

  return value;
}

module.exports.iterate = iterate;
module.exports.fizzBuzz = fizzBuzz;